src/rgw/rgw_auth_s3.cc

```diff
@@ -1022,6 +1022,32 @@ size_t AWSv4ComplMulti::recv_body(char* const buf, const size_t buf_max)
   return buf_pos;
 }
 
+size_t AWSv4ComplMulti::recv_body(char* const buf, const size_t buf_max)
+{
+  /* get total buf_max data from the stream, buf_max's default is  rgw_max_chunk_size
+   * we're getting from a client. */
+  size_t buf_pos = 0; 
+  size_t buf_incomplement_size = buf_max; //bul_max is 4M, exclude the last muiltpart
+
+  size_t len  = 0;
```

Comment:
dont think you need `len`, just use `read_len` inside the loop

Reply:
Yes, There is no need for len. Thank you for time and consideration.